site/src/pages/docs/advanced-spring-boot-integration.mdx

```diff
@@ -0,0 +1,400 @@
+
+    // Automatically injected by Spring
+    @Autowired
+    public TodoAnnotatedService(TodoRepository todoRepository) {
+        this.todoRepository = todoRepository;
+    }
+
+    @Get("/:id")
+    public HttpResponse get(@Param Integer id) {
+        Todo todo = todoRepository.get(id);
+        if (todo == null) {
+            HttpResponse.of(HttpStatus.NO_CONTENT);
+        }
+        return HttpResponse.ofJson(todo);
+    }
+
+    @Post
+    public HttpResponse create(Todo todo) {
+        final int result = todoRepository.create(todo);
+        if (result == 0) {
+            return HttpResponse.of(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        return HttpResponse.of(HttpStatus.CREATED);
+    }
+}
+```
+
+Inject this service to the <type://ArmeriaServerConfigurator> bean.
+You can serve both legacy Spring service and Armeria service at the same port.
+
+```java filename=ArmeriaConfiguration.java
+@Bean
+public ArmeriaServerConfigurator armeriaServerConfigurator(TomcatService tomcatService, TodoAnnotatedService todoAnnotatedService) {
+   return serverBuilder -> {
+       serverBuilder
+               .serviceUnder("/", tomcatService)
+               .annotatedService("/armeria", todoAnnotatedService);
+   };
+}
+```
+
+## Internal Services
+
+Armeria supports four internal services which are helpful for monitoring and management.
+You can use those services by include the service ids to application.yml (or application.properties).
+Service ids are `docs`, `health`, `metrics`, `actuator` or you can use them `all`.
+
+```yaml filename=application.yml
+armeria:
+  ports:
+    - port: 8080
+      protocols:
+        - HTTP
+  # Add the below configuration
+  internal-services:
+    include: docs, health, metrics, actuator  # you can add all of the services using `all`
+    port: 8090
+```
+
+The `armeria.internal-services.port` is unnecessary. If the port is not specified or left as 0,
+it will bind the internal service to a random unused port.
+It is ok to set the port of internal service same as one of `armeria.ports`.
+
+<Tip>
+
+  The internal services described below must be configured through the beans described in the documentation.
+  Do not use the method of <type://ServerBuilder> directly using <type://ArmeriaServerConfigurator> or
+  `Consumer<ServerBuilder>` bean. They may be overwritten by the automatic configuration of Armeria.
+
+</Tip>
+
+<Tip>
+
+  If you have configured the server as described in the preceding section of the document,
+  it will also serve on the internal service port. If you want to separate port,
+  use <type://ServerBuilder#defaultVirtualHost()> or <type://ServerBuilder#virtualHost(int)>.
+
+```java
+@Bean
+public ArmeriaServerConfigurator armeriaServerConfigurator(TomcatService tomcatService) {
+    return sb -> sb
+            .defaultVirtualHost()  // Add this method.
+            .serviceUnder("/", tomcatService)
+            .and();
+}
+```
+
+</Tip>
```

Comment:
```suggestion
```
I remove this tip because the document service can only create documents for the virtual host it belongs to. Therefore, if user applies above tip, user may experience difficulties when using the document service. Also, It seems that the situation mentioned in the tip is not important use case.